Monika-After-Story/MonikaModDev
# New greeting, new clothes and hair, and hair sprite adjustments

Main goal is to add a greeting which adds more life.

### Changes:
- Added a new greeting where Monika returns from shower or bath.
- Added new hairstyle (wet hair), new outfit (bath towel), and an "accessory" (water drops on Monika's body).  Thanks @kkrosie123 for helping me fix them!
- Added new static class `MASTimedeltaRepeatRule`. A rule for events we want to repeat only after certain amount of time has passed since last seen. It works for both farewells and greetings.
- ~~Deleted some old night sprites + removed white pixels from some hairstyles (def, ponytailbraid, twintails, usagi).~~
- ~~Slightly touched up the color of the hair sprites so they're less red~~
- ~~Added the hair spritepack in `Orcaramelo Sakuya` to the `Orca Hairs` folder~~
- Added several new function: `mas_sprites.get_installed_sprites` which takes a type constant and a predicate function, returns sprite objects. And more specific functions: `mas_sprites.get_installed_acs`, `mas_sprites.get_installed_hair`, and `mas_sprites.get_installed_clothes`.

### Testing:
- Verify the greetings can happen only once in 3 days.
- Verify the new hair, outfit, and accessory look good and work as expected. Note: those aren't selectable by user at least for now, since it doesn't make sense. We don't need weird combinations like the blazer + wet hair.
- Verify the cleanup event is being run appropriately. That's being after its `start_date` passed during game sesh and during the time user being absent.
- ~~Verify the adjusted hairstyles look good~~
- ~~Verify we removed all unused night sprites.~~
- ~~Verify adjusted hair looks good~~
- ~~Verify copied hair spritepack works properly.~~
- Verify the new `mas_sprites.get_installed_*` functions work.
